### ISSUE TITLE
Toc: Highlighting the current section in the contents menu

### ISSUE THREAD

**Issue by F-A** [ipython-contrib/jupyter_contrib_nbextensions#944]
Hi all,

first of all: thank you for the nice work on nbextensions! They are really useful.

I was wondering: would it be possible to highlight the currently displayed section in the table of contents window as we scroll the notebook?
This would be really improve the navigation in long notebooks with "redundant" code/cells.

Chers,
Florian

**Comment by jcb91** [comment]
Yep, sounds possible. When several headings are visible at once, should they all be highlighted in the TOC, or just the first visible one/last non-visible one?

**Comment by F-A** [comment]
I'm not sure what the perfect behaviour would be.
Maybe highlight the headers which corresponds to the first cell displayed in the notebook (even if this header is not visible)?

**Comment by cqcn1991** [comment]
@jcb91 I tried this before, ideally, you can just plug ` Scrollspy from Bootstrap` into `toc2`, see http://getbootstrap.com/javascript/#scrollspy for docs.

However, when I was working on toc2, it was not possible, the problem is the Jupyter Notebook itself. While we can scroll it, the css is actually fixed, see https://github.com/jupyter/notebook/issues/1202

So if the Jupyter notebook has fixed the position problem, we can have it done in minutes. If not, you may need some hacking.

**Comment by F-A** [comment]
The issue jupyter/notebook#1202 was closed recently. Does this mean that it is fixed and this toc2 enhancement is possible?

**Comment by cqcn1991** [comment]
@F-A no, it simply means no one cares for this problem right now, and there is no existing solution for that css problem. You can open it again if you like.

**Comment by jcb91** [comment]
@cqcn1991 I think the scrollspy should actually work just fine. It doesn't need to be applied to the `body` element, but the `#site` div, since this is the one which actually scrolls...

**Comment by cqcn1991** [comment]
@jcb91 really? That would great!

**Comment by jcb91** [comment]
That's certainly how it looks to me - from [bootstrap docs](http://getbootstrap.com/javascript/#requires-relative-positioning):

> No matter the implementation method, scrollspy requires the use of position: relative; on the element you're spying on. In most cases this is the `<body>`. When scrollspying on elements other than the `<body>`, be sure to have a height set and `overflow-y: scroll;` applied.

The `#site` div is the one with a defined height and `overflow: auto`, so I think you want something along the lines of 

```javascript
$('#site').css('overflow-y', 'scroll').scrollspy({ target: '#toc' })
```

though I haven't tested it (and you'd need to make all the toc links into bootstrap navs, again as per scrollspy docs).
